File: collect_app/src/main/java/org/odk/collect/android/notifications/Notifier.java
package org.odk.collect.android.notifications;

import org.odk.collect.android.formmanagement.ServerFormDetails;
import org.odk.collect.forms.FormSourceException;

import java.util.List;
import java.util.Map;

public interface Notifier {

    void onUpdatesAvailable(List<ServerFormDetails> updates);

    void onUpdatesDownloaded(Map<ServerFormDetails, String> result);

    void onSync(FormSourceException exception);

    void onSubmission(boolean failure, String message);
}

File: collect_app/src/main/java/org/odk/collect/android/notifications/Notifier.kt
package org.odk.collect.android.notifications

import org.odk.collect.android.formmanagement.ServerFormDetails
import org.odk.collect.android.formmanagement.download.FormDownloadException
import org.odk.collect.android.instancemanagement.InstanceUploadResult
import org.odk.collect.forms.FormSourceException
import org.odk.collect.forms.instances.Instance

interface Notifier {
    fun onUpdatesAvailable(updates: List<ServerFormDetails>, projectId: String)
    fun onUpdatesDownloaded(result: Map<ServerFormDetails, FormDownloadException?>, projectId: String)
    fun onSync(exception: FormSourceException?, projectId: String)
    fun onSyncStopped(projectId: String)
    fun onSubmission(uploadResults: List<InstanceUploadResult>, projectId: String)
}
